cloud/docker_client.go

```diff
@@ -125,46 +119,47 @@ func (c *dockerClientImpl) CreateContainer(ctx context.Context, id string, d dis
 
 	grip.Info(message.Fields{
 		"message":       "Creating docker container",
-		"name":          id,
+		"name":          name,
 		"image_id":      containerConf.Image,
 		"exposed_ports": containerConf.ExposedPorts,
 		"port_bindings": hostConf.PortBindings,
 	})
 
 	// Build container
-	if _, err := dockerClient.ContainerCreate(ctx, containerConf, hostConf, networkConf, id); err != nil {
-		err = errors.Wrapf(err, "Docker create API call failed for container '%s'", id)
+	if _, err := dockerClient.ContainerCreate(ctx, containerConf, hostConf, networkConf, name); err != nil {
+		err = errors.Wrapf(err, "Docker create API call failed for container '%s'", name)
 		grip.Error(err)
 		return err
 	}
 
 	return nil
 }
 
-// GetContainer returns low-level information on the Docker container.
-func (c *dockerClientImpl) GetContainer(ctx context.Context, h *host.Host) (*types.ContainerJSON, error) {
-	dockerClient, err := c.generateClient(h.Distro)
+// GetContainer returns low-level information on the Docker container with the
+// specified ID running on the specified host machine.
+func (c *dockerClientImpl) GetContainer(ctx context.Context, h *host.Host, id string) (*types.ContainerJSON, error) {
+	dockerClient, err := c.generateClient(h)
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to generate docker client")
 	}
 
-	container, err := dockerClient.ContainerInspect(ctx, h.Id)
+	container, err := dockerClient.ContainerInspect(ctx, id)
 	if err != nil {
-		return nil, errors.Wrapf(err, "Docker inspect API call failed for container '%s'", h.Id)
+		return nil, errors.Wrapf(err, "Docker inspect API call failed for container '%s'", id)
 	}
 
 	return &container, nil
 }
 
-// ListContainers lists all containers running on the distro-specified host machine.
-func (c *dockerClientImpl) ListContainers(ctx context.Context, d distro.Distro) ([]types.Container, error) {
-	dockerClient, err := c.generateClient(d)
+// ListContainers lists all containers running on the specified host machine.
+func (c *dockerClientImpl) ListContainers(ctx context.Context, h *host.Host) ([]types.Container, error) {
+	dockerClient, err := c.generateClient(h)
 	if err != nil {
 		return nil, errors.Wrap(err, "Failed to generate docker client")
 	}
 
-	// Get all the things!
-	opts := types.ContainerListOptions{All: true}
+	// Get all running containers
+	opts := types.ContainerListOptions{All: false}
```

Comment:
Set all to false because it makes sense that ListContainers should return only running containers (All = true returns terminated containers as well). Should I instead pass a types.ContainerListOptions object as a parameter to ListContainers to be able to customize this behavior?